### PR TITLE
Feature/disconnect initialize simulants from event system

### PR DESCRIPTION
**Pull request by collijk**

### CODE REVIEW

File: vivarium/framework/population.py [hunk 1]
```diff
@@ -178,21 +190,60 @@ def get_view(self, columns, query=None):
...
 
-    def _create_simulants(self, count, population_configuration=None):
+    def _order_initializers(self) -> None:
+        unordered_initializers = deque(self._population_initializers)
+        starting_length = -1
+        available_columns = []
+        self._population_initializers = []
+
+        # This is the brute force N^2 way because constructing a dependency graph is work
+        # and in practice this should run in about order N time due to the way dependencies are
```
Comment (alec-deason) [inline vivarium/framework/population.py]: Also, N is always meaningfully small

Comment (c-werner) [reply vivarium/framework/population.py]: Should it be an error for two initializers to create the same columns?

File: vivarium/framework/population.py [hunk 2]
```diff
@@ -178,21 +190,60 @@ def get_view(self, columns, query=None):
         """
         return PopulationView(self, columns, query)
 
-    def get_simulant_creator(self):
+    def register_simulant_initializer(self, initializer: Callable,
+                                      creates_columns: Sequence[str]=(), requires_columns: Sequence[str]=()):
+        self._population_initializers.append((initializer, creates_columns, requires_columns))
```
Comment (c-werner) [inline vivarium/framework/population.py]: should this `raise` if `_initializers_ordered`?

Comment (collijk) [reply vivarium/framework/population.py]: There's an implicit contract that there won't be `_initializers_ordered` when this method gets called.  This method is only accessible from the `Builder` object that gets passed into each component's setup method.  `_initializers_ordered` only gets populated the first time we create simulants, an event guaranteed by the framework to occur after the setup phase.

File: vivarium/framework/population.py (hunk 1 above)
Comment (collijk) [reply vivarium/framework/population.py]: Yes.  You can see below (Line 223) that makes this check.  We had to comment it out to deal with an issue in a downstream library we use internally.  It'll be updated soon.

**Pull request merged**